lnst/RecipeCommon/Perf/Measurements/RDMABandwidthMeasurement.py

```diff
@@ -0,0 +1,135 @@
+from dataclasses import dataclass
+from typing import Any, Optional
+import time
+
+from lnst.Common.IpAddress import BaseIpAddress
+from lnst.Controller.Host import Host
+from lnst.Controller.Job import Job
+from lnst.Controller.Recipe import BaseRecipe
+from lnst.Controller.RecipeResults import ResultType
+from lnst.Tests.RDMABandwidth import RDMABandwidthServer, RDMABandwidthClient
+from lnst.RecipeCommon.Perf.Measurements.BaseMeasurement import BaseMeasurement
+from lnst.RecipeCommon.Perf.Measurements.Results import (
+    RDMABandwidthMeasurementResults,
+    AggregatedRDMABandwidthMeasurementResults
+)
+
+
+@dataclass
+class RDMAFlow:
+    generator: Host
+    receiver: Host
+    generator_nic_name: str
+    receiver_nic_name: str
+    generator_bind: BaseIpAddress
+    receiver_bind: BaseIpAddress
+
+
+@dataclass
+class RDMAFlowTest:
+    flow: RDMAFlow
+    server_job: Job
+    client_job: Job
+
+
+class RDMABandwidthMeasurement(BaseMeasurement):
+    def __init__(
+        self,
+        flows: list[RDMAFlow],
+        recipe_conf: Any = None,
+    ):
+        super().__init__(recipe_conf)
+
+        self._flows = flows
+        self._running_flow_tests: list[RDMAFlowTest] = []
+        self._finished_flow_tests: list[RDMAFlowTest] = []
+
+    @property
+    def flows(self) -> list[RDMAFlow]:
+        return self._flows
+
+    def start(self) -> None:
+        self._running_flow_tests = self._prepare_flow_tests()
+
+        for flow_test in self._running_flow_tests:
+            flow_test.server_job.start(bg=True)
+
+        time.sleep(2)
+
+        for flow_test in self._running_flow_tests:
+            flow_test.client_job.start(bg=True)
+
+    def finish(self) -> None:
+        for flow_test in self._running_flow_tests:
+            flow_test.client_job.wait()
+            flow_test.server_job.wait()
+
+        self._finished_flow_tests = self._running_flow_tests
+        self._running_flow_tests = []
+
+    def collect_results(self) -> list[RDMABandwidthMeasurementResults]:
+        results: list[RDMABandwidthMeasurementResults] = []
+        for flow_test in self._finished_flow_tests:
+            results.append(
+                RDMABandwidthMeasurementResults(
+                    measurement=self,
+                    bandwidth=flow_test.client_job.result["bandwidth"],
```

Comment:
Maybe we should include also the server_job results here. But I'm not familiar with the tool and do not know if it generates anything.

Reply:
server result == client result == single float bandwidth `MiB/s` result.